pet-store-react-hooks/src/infrastructure/components/Modal/Modal.tsx

```diff
@@ -13,25 +11,34 @@ interface ModalProps {
 export function Modal(props: ModalProps) {
   const { children, onBackdropClick } = props;
 
-  useEffect(() => {
-    const handleBackdropClick = (event: MouseEvent) => {
-      if (modalRoot && modalRoot === event.target) {
+  const modalBackdropRef = useRef<HTMLDivElement>(null);
+
+  // Question: With this function and useEffect() there should be a simpler way for handling
+  const handleBackdropClick = useCallback(
```

Comment:
Why are you doing all this? Just add the `onClick` handler on the backdrop directly.